pkg/controller/user-cluster-controller-manager/resources/reconciler.go

```diff
@@ -655,3 +663,13 @@ type reconcileData struct {
 	userSSHKeys   map[string][]byte
 	cloudConfig   []byte
 }
+
+func (r *reconciler) ensureOPAIntegrationIsRemoved(ctx context.Context) error {
+	for _, resource := range gatekeeper.GetResourcesToRemoveOnDelete() {
+		if err := r.Client.Delete(ctx, resource); err != nil && !errors.IsNotFound(err) {
```

Comment:
You might also need to check if the NotFound error is reported

Reply:
Its checking for NotFound already, or did you mean something else?

Reply:
Yes you are right, sorry I overlooked that